planner/core/exhaust_physical_plans.go

```diff
@@ -1264,9 +1306,36 @@ func (la *LogicalAggregation) getHashAggs(prop *property.PhysicalProperty) []Phy
 }
 
 func (la *LogicalAggregation) exhaustPhysicalPlans(prop *property.PhysicalProperty) []PhysicalPlan {
-	aggs := make([]PhysicalPlan, 0, len(la.possibleProperties)+1)
-	aggs = append(aggs, la.getHashAggs(prop)...)
-	aggs = append(aggs, la.getStreamAggs(prop)...)
+	preferHash := (la.preferAggType & preferHashAgg) > 0
+	preferStream := (la.preferAggType & preferStreamAgg) > 0
+	if preferHash && preferStream {
+		errMsg := "Optimizer aggregation hints are conflicted"
```

Comment:
How about extract a common error message pattern for conflicted optimizer hints. For example:
```go
errMsgPattern := "Conflicted SQL Hints for %s. You can only specify one of these hints: %s"
errConflictedSQLHints := terror.ClassOptimizer.New(codeInternal, errMsgPattern)
```

The above error can be used this way:
```go
errConflictedSQLHints.GenWithStackByArgs(
	"Aggregate",
	strings.Join([]string{"TIDB_HASHAGG()", "TIDB_STREAMAGG()"}, ","),
)
```

Example usage for Join:
```go
errConflictedSQLHints.GenWithStackByArgs(
	"Aggregate",
	strings.Join([]string{"TIDB_HJ()", "TIDB_SMJ()"}, ","),
)
```

You can improve the above error message by replacing `TIDB_HASHAGG()` with the string extracted from the AST node, which is written by the user.

Reply:
I think we can do this later, in another PR.

Reply:
Okay